module/VuFind/tests/unit-tests/src/VuFindTest/ChannelProvider/RandomTest.php

```diff
@@ -0,0 +1,178 @@
+<?php
+/**
+ * Random Test Class
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2022.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Sudharma Kellampalli <skellamp@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+namespace VuFindTest\ChannelProvider;
+
+use VuFind\ChannelProvider\Random;
+use VuFindSearch\ParamBag;
+use VuFindSearch\Query\Query;
+use VuFindTest\RecordDriver\TestHarness;
+
+/**
+ * Random Test Class
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Sudharma Kellampalli <skellamp@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+class RandomTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * Test deriving channel information from a record driver object.
+     *
+     * @return void
+     */
+    public function testGetFromRecord(): void
+    {
+        $parameters = $this->supportMethod();
+        $random = $parameters[0];
+        $recordDriver = $this->getMockBuilder(\VuFind\RecordDriver\AbstractBase::class)
+            ->getMock();
+        $recordDriver->expects($this->once())->method('getSourceIdentifier')
+            ->willReturn('Solr');
+        $this->assertSame($parameters[1], $random->getFromRecord($recordDriver));
+    }
+
+    /**
+     * Test deriving channel information from a search results object.
+     *
+     * @return void
+     */
+    public function testGetFromSearch(): void
+    {
+        $results = $this->getMockBuilder(\VuFind\Search\Base\Results::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $parameters = $this->supportMethod();
+
+        $results->expects($this->once())->method('getParams')
+            ->willReturn($parameters[2]);
+
+        $random = $parameters[0];
+        $this->assertSame($parameters[1], $random->getFromSearch($results));
+    }
+
+    /**
+     * Support method to mock objects.
+     *
+     * @return array
+     */
+    public function supportMethod()
+    {
```

Comment:
This method is quite long, and it might be a good idea to break it into smaller parts with the help of support methods. I don't insist on this -- it's only a test class, and I don't want to slow you down from moving on to other work -- but in principle, smaller methods are easier to follow and make it easier to see how the variables relate to one another. As a first step, simply factoring out all the setup of various mock objects might help -- e.g., at the end, you could probably change:

```
        $coverRouter = $this->getMockBuilder(\VuFind\Cover\Router::class)
            ->disableOriginalConstructor()
            ->getMock();
        $coverRouter->expects($this->once())->method('getUrl')
            ->with($this->equalTo($recordDriver), $this->equalTo('medium'))
            ->willReturn('foo_Thumbnail');
        $recordRouter = $this->getMockBuilder(\VuFind\Record\Router::class)
            ->disableOriginalConstructor()
            ->getMock();
        $recordRouter->expects($this->once())->method('getTabRouteDetails')
            ->with($this->equalTo($recordDriver))
            ->willReturn('foo_Route');
        $random->setCoverRouter($coverRouter);
        $random->setRecordRouter($recordRouter);```

to

```
    protected function getConfiguredCoverRouterMock($recordDriver)
    {
        $coverRouter = $this->getMockBuilder(\VuFind\Cover\Router::class)
            ->disableOriginalConstructor()
            ->getMock();
        $coverRouter->expects($this->once())->method('getUrl')
            ->with($this->equalTo($recordDriver), $this->equalTo('medium'))
            ->willReturn('foo_Thumbnail');
        return $coverRouter;
    }

    protected function getConfiguredRecordRouterMock($recordDriver)
    {
        $recordRouter = $this->getMockBuilder(\VuFind\Record\Router::class)
            ->disableOriginalConstructor()
            ->getMock();
        $recordRouter->expects($this->once())->method('getTabRouteDetails')
            ->with($this->equalTo($recordDriver))
            ->willReturn('foo_Route');
        return $recordRouter;
    }

...

        $random->setCoverRouter($this->getConfiguredCoverRouterMock($recordDriver));
        $random->setRecordRouter($this->getConfiguredRecordRouterMock($recordDriver));
```

(And of course there are other segments of the method that could be given similar treatment; that's just one example).